### PR TITLE
Clean UML and Document PlantUML Setup for IntelliJ

### PR DESCRIPTION
**Pull request by bruderol**
just a little bit clean up of the plantuml uml class diagram (more logic order) - Long live "Clean UML" :-)

Also documented how to setup the plantUML stuff to wrok in IntelliJ

### CODE REVIEW

**Comment by lynparbone** [comment]
Note: To display the graphs, I've installed Graphviz and set the GRAPHVIZ_DOT environment variable appropriately. I also installed the plantuml framework in IntelliJ (see [here](https://www.jetbrains.com/help/idea/markdown.html#diagrams)). 
Oddly, the uml diagrams aren't displayed in my IntelliJ editor "as is" (even though the puml code is now embedded, see #1009 ). In my markdown, I receive the error: No dot exectuable found. Cannot find Graphviz.
However, if I change anything in the UML code (e.g. edit the notes), the diagrams are displayed.  Also new diagrams I add myself are displayed appropriately. So apparently it does work.... ;-)

Should we add the installation to the scenarioo documentation as a requirement to work with plantuml in IntelliJ?

File: docs/features/details/Details.md [hunk 1]
```diff
@@ -17,8 +17,6 @@ class UseCase
 class Scenario
```
Comment (lynparbone) [inline docs/features/details/Details.md:17]: Add the _details_ attribute to the _Branch, Build, UseCase, Scenario, Step_ classes

File: docs/features/details/Details.md [hunk 2]
```diff
@@ -29,47 +27,58 @@ An object's further detail properties as entries with
                  e.g. a simple string or a more complex object
 end note
 
+Branch --> Details : details
```
Comment (lynparbone) [inline docs/features/details/Details.md]: Add multiplicity (1) to associations of _Branch, Build, UseCase, Scenario, Step_

File: docs/features/details/Details.md [hunk 3]
```diff
@@ -29,47 +27,58 @@ An object's further detail properties as entries with
...
+
+
 class ObjectDescription {
     name: String
     type: String
     details: Details
 }
 
+Object <|-- ObjectDescription
+
```
Comment (lynparbone) [inline docs/features/details/Details.md:52]: The association from _ObjectDescription_ to _Details_ is missing.
Or is it omitted on purpose so the diagram isn't too cluttered?

File: docs/features/details/Details.md [hunk 4]
```diff
@@ -29,47 +27,58 @@ An object's further detail properties as entries with
...
-
-Object <|-- String
-Object <|-- ObjectDescription
-Object <|-- ObjectReference
-Object <|-- ObjectList
 Object <|-- ObjectTreeNode
-
-ObjectList --> "*" Object : items
 ObjectTreeNode --> "1" Object : item
 ObjectTreeNode --> "*" ObjectTreeNode : children
```
Comment (lynparbone) [inline docs/features/details/Details.md:79]: The association from _ObjectTreeNode_ to _Details_ is missing.
Or is it omitted on purpose so the diagram isn't too cluttered?

**Review by lynparbone — COMMENTED** [review]
I added a few comments to individual lines regarding small UML details that could be added. 
The general structuring is good!

File: docs/features/details/Details.md (hunk 2 above)
Comment (bruderol) [reply docs/features/details/Details.md]: I find this not so important - If I not use any multiplicity it means always (1) - this just clutters the UML diagrams unnecessarily IMHO.

File: docs/features/details/Details.md (hunk 3 above)
Comment (bruderol) [reply docs/features/details/Details.md:52]: the last sentence is true for your other two comments ;-) But for this one I am not sure ;-)

File: docs/features/details/Details.md (hunk 2 above)
Comment (bruderol) [reply docs/features/details/Details.md]: as discussed I will try to add it none the less

File: docs/features/details/Details.md (hunk 4 above)
Comment (bruderol) [reply docs/features/details/Details.md:79]: as discussed - yes not to clutter

File: docs/features/details/Details.md (hunk 3 above)
Comment (bruderol) [reply docs/features/details/Details.md:52]: yes., here as well

File: docs/features/details/Details.md (hunk 1 above)
Comment (bruderol) [reply docs/features/details/Details.md:17]: the dependency line is enough here

File: docs/features/details/Details.md (hunk 2 above)
Comment (bruderol) [reply docs/features/details/Details.md]: done

File: docs/features/details/Details.md (hunk 3 above)
Comment (bruderol) [reply docs/features/details/Details.md:52]: we decided to ommit on purpose

File: docs/features/details/Details.md (hunk 4 above)
Comment (bruderol) [reply docs/features/details/Details.md:79]: we decided to ommit on purpose

**Comment by bruderol** [comment]
@lynparbone yes, i will add a short notice in the developer docu how to use it.

**Comment by bruderol** [comment]
@lynparbone I documented the setup - I think we can merge that now - agree?

**Pull request merged**